bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/utilities/ICloudTextTranslator.java

```diff
@@ -37,11 +39,13 @@ public ICloudTextTranslator(Bundle bundle, TranslationProvider i18nProvider, Loc
     }
 
     public String getText(String key, Object... arguments) {
-        Locale locale = localeProvider != null ? localeProvider.getLocale() : Locale.ENGLISH;
-        return i18nProvider != null ? i18nProvider.getText(bundle, key, getDefaultText(key), locale, arguments) : key;
+        Locale locale = localeProvider.getLocale();
+        String retText = i18nProvider.getText(bundle, key, getDefaultText(key), locale, arguments);
+        return retText != null ? retText : key;
     }
 
     public String getDefaultText(@Nullable String key) {
-        return i18nProvider.getText(bundle, key, key, Locale.ENGLISH);
+        String retText = i18nProvider.getText(bundle, key, key, Locale.ENGLISH);
+        return retText != null ? retText : key != null ? key : "UNKNOWN_TEXT";
```

Comment:
Any reason for key to be nullable? I believe this logic could be simplified, since it doesn't make sense to call this method without a valid argument.

Reply:
The initial code wasn't written by be and the change is from @lsiepel . I guess is nullable, because it is also in the called TranslationProvider:

```
@NonNullByDefault
public interface TranslationProvider {

    /**
     * Returns a translation for the specified key in the specified locale (language) by only
     * considering the translations within the specified module.
     * <p>
     * If no translation could be found, the specified default text is returned.<br>
     * If the specified locale is {@code null}, the default locale is used.
     *
     * @param bundle the module to be used for the look-up (could be null)
     * @param key the key to be translated (could be null or empty)
     * @param defaultText the default text to be used (could be null or empty)
     * @param locale the locale (language) to be used (could be null)
     * @return the translated text or the default text (could be null or empty)
     */
    @Nullable
    String getText(@Nullable Bundle bundle, @Nullable String key, @Nullable String defaultText,
            @Nullable Locale locale);
```